link/link.go

```diff
@@ -119,7 +119,7 @@ func wrapRawLink(raw *RawLink) (_ Link, err error) {
 	case UprobeMultiType:
 		return &uprobeMultiLink{*raw}, nil
 	case PerfEventType:
-		return nil, fmt.Errorf("recovering perf event fd: %w", ErrNotSupported)
+		return &perfEventLink{*raw, nil}, nil
```

Comment:
How do we know this is `perfEventLink` and not `perfEventIoctl`?

Also, `perfEventLink.PerfEvent` doesn't check for `pe == nil` right now and crashes.